src/msg/simple/Pipe.cc

```diff
@@ -2440,8 +2453,11 @@ int Pipe::write_message(const ceph_msg_header& header, const ceph_msg_footer& fo
 
 int Pipe::tcp_read(char *buf, unsigned len)
 {
-  if (sd < 0)
-    return -1;
+  int rc = 0;
+  if (sd < 0) {
+    rc = -errno;
```

Comment:
errno isn't getting set by anything - it'll just have an unrelated value here.

For these sanity checks, tcp_{read,write} should return a specific error, perhaps EINVAL